### ISSUE TITLE
Make @Fallback "Istio-aware"

### ISSUE THREAD

**Issue by lordofthejars** [snowdrop/spring-fallback#2]
When circuit is opened in Istio, the connection returns an Http status code 503. This means that everytime I do a connection I will need to do something like:

```java
try {
  connection();
} catch (ConnectionException e) {
  if (e.getStatus() == 503) {
   CB opened
  } else {
    other error (404, ...);
  }
}
```
This is repeated in every place. So to make more Istio aware, we should think about creating an annotation `@RestTemplateIstioFallback` (that meta-annotates `@Fallback)` and automatically checks the error code.

Maybe this is too specific since now we are assuming that only `RestTemplate` is used, but if we create a correct model where we can create more custom IstioFallbacks then we will be able to extend to other clients.

**Comment by geoand** [comment]
I like the idea a lot.

Perhaps we should create a different maven module that will create more Istio specific stuff?
I mean have the basic in a core module and then add an Istio module.

**Comment by lordofthejars** [comment]
Yes it is just an idea, I have not thought a lot of internal details, but I know tthat how the status code is retrieved will depend on the underneath framework used (now since it is spring boot it might be easier) but there is grpc or wy not extend to microprofile client/jaxrs.

**Comment by geoand** [comment]
Sounds good to me :)
